### PR TITLE
[ADI] Manage ADI while unregistration + filter threads by answer id

### PR DESCRIPTION
**Pull request by illusionOfParadise**

### CODE REVIEW

File: features/api/ideas_workshop_threads.feature [hunk 1]
```diff
@@ -144,149 +144,115 @@ Feature:
...
-                "created_at":"@string@.isDateTime()",
-                "approved": false
+                "approved": false,
+                "created_at": "@string@.isDateTime()"
             }
         ]
     }
     """
 
   Scenario: As a non logged-in user I can see visible threads for an idea
```
Comment (JeremieSamson) [inline features/api/ideas_workshop_threads.feature]:
```suggestion
  Scenario: As a non logged-in user I can see visible threads for an answer
```

File: src/Repository/IdeaRepository.php [hunk 2]
```diff
@@ -108,4 +109,35 @@ public function getAdherentVotesForIdea(Idea $idea, Adherent $adherent): array
             return $result;
         }, []);
     }
+
+    public function removeNotFinalizedIdeas(Adherent $author)
```
Comment (JeremieSamson) [inline src/Repository/IdeaRepository.php]: shouldn't we add void typehint here ?

File: src/Repository/IdeaRepository.php [hunk 3]
```diff
@@ -108,4 +109,35 @@ public function getAdherentVotesForIdea(Idea $idea, Adherent $adherent): array
...
+            ->where('idea.author = :author')
+            ->andWhere('idea.status != :status')
+            ->setParameter('author', $author)
+            ->setParameter('status', IdeaStatusEnum::FINALIZED)
+            ->getQuery()
+            ->execute()
+        ;
+    }
+
+    public function anonymizeFinalizedIdeas(Adherent $author)
```
Comment (JeremieSamson) [inline src/Repository/IdeaRepository.php]: same here

File: src/Repository/AuthorTrait.php [hunk 4]
```diff
@@ -0,0 +1,20 @@
+<?php
+
+namespace AppBundle\Repository;
+
+use AppBundle\Entity\Adherent;
+
+trait AuthorTrait
+{
+    public function removeAuthorItems(Adherent $author)
```
Comment (ottaviano) [inline src/Repository/AuthorTrait.php]: can you add typehint pls

File: src/Repository/IdeaRepository.php (hunk 2 above)
Comment (ottaviano) [inline src/Repository/IdeaRepository.php]: typehint

File: src/Repository/IdeaRepository.php (hunk 3 above)
Comment (ottaviano) [inline src/Repository/IdeaRepository.php]: typehint

File: src/Repository/IdeaRepository.php [hunk 5]
```diff
@@ -108,4 +109,35 @@ public function getAdherentVotesForIdea(Idea $idea, Adherent $adherent): array
...
+    {
+        $qb = $this->createQueryBuilder('idea');
```
Comment (ottaviano) [inline src/Repository/IdeaRepository.php]: you can avoid creating `$qb` variable here

File: src/Repository/IdeaRepository.php [hunk 6]
```diff
@@ -108,4 +109,35 @@ public function getAdherentVotesForIdea(Idea $idea, Adherent $adherent): array
...
+
+        $qb->update()
+            ->set('idea.author', ':new_value')
```
Comment (ottaviano) [inline src/Repository/IdeaRepository.php]: what about : `->set('idea.author', null)`

**Pull request merged**